packages/dds/tree/src/feature-libraries/treeTextCursor.ts

```diff
@@ -116,17 +118,24 @@ export class TextCursor implements ITreeCursor {
     }
 
     up(): TreeNavigationResult {
-        if (this.parentStack.length === 0) {
+        const length = this.parentStack.length;
+        assert(this.indexStack.length === length, "Unexpected indexStack.length");
```

Comment:
Have you considered merging the stacks into one so that this kind of issue is checked by the type system?

Reply:
Yes.

I replicated the style of @DLehenbauer 's json cursor as that apparently had undergone perf testing.

The idea with this approach is to avoid allocating an object every time we navigate the cursor down.